components/feature/addons/src/test/java/mozilla/components/feature/addons/ui/AddonInstallationDialogFragmentTest.kt

```diff
@@ -165,6 +172,32 @@ class AddonInstallationDialogFragmentTest {
         }
     }
 
+    @Test
+    fun `allows state loss when committing`() {
+        val addon = mock<Addon>()
+        val mockedCollectionProvider = mock<AddonCollectionProvider>()
+        val fragment = createAddonInstallationDialogFragment(addon, mockedCollectionProvider)
+
+        val fragmentManager = mock<FragmentManager>()
+        val fragmentTransaction = mock<FragmentTransaction>()
+        `when`(fragmentManager.beginTransaction()).thenReturn(fragmentTransaction)
+
+        fragment.show(fragmentManager, "test")
+        verify(fragmentTransaction).commitAllowingStateLoss()
```

Comment:
Not sure if this is the right place, but should we verify that `addonCollectionProvider` is assigned and not null?

Reply:
I don't think this test is good for covering that, but I added some assertions to the `build dialog` test to cover it now.